src/topology/continuous_function/basic.lean

```diff
@@ -161,6 +169,51 @@ instance [linear_order β] [order_closed_topology β] : lattice C(α, β) :=
 
 -- TODO transfer this lattice structure to `bounded_continuous_function`
 
+section sup'
+variables [linear_order γ] [order_closed_topology γ]
+
+@[simp]
+lemma sup'_coe {ι : Type*} {s : finset ι} (H : s.nonempty) (f : ι → C(β, γ)) :
+  ((s.sup' H f : C(β, γ)) : ι → β) = s.sup' H (λ a, (f a : β → γ)) :=
+begin
+  classical,
+  revert H,
+  apply finset.cons_induction_on s,
+  { rintro ⟨a, ⟨⟩⟩, },
+  { rintros a s' nm ih n',
+    by_cases n : s'.nonempty,
+    { rw [finset.sup'_cons n, finset.sup'_cons n],
+      simp [sup_coe, ih n], },
+    { have e : s' = ∅ := finset.not_nonempty_iff_eq_empty.mp n,
+      subst e,
+      refl, }, },
+end
```

Comment:
This proof (and also `finset.sup'_apply`) would be much shorter if we had a `comp_sup'_eq_sup'_comp` lemma to match `comp_sup_eq_sup_comp`.

Reply:
Something like:
```
lemma comp_sup'_eq_sup'_comp [semilattice_sup β] [semilattice_sup γ] {s : finset α} (H : s.nonempty)
  (f : α → β) (g : β → γ) (hg : ∀ i j, g (i ⊔ j) = g i ⊔ g j) :
  g (s.sup' H f) = s.sup' H (λ a, g (f a)) :=
begin
  rw [←with_bot.coe_eq_coe, coe_sup'],
  let g' := @with_bot.rec_bot_coe (β) (λ _, with_bot (γ)) ⊥ (λ x, ↑(g x)),
  show g' ↑(s.sup' H f) = s.sup (λ a, g' ↑(f a)),
  rw coe_sup',
  refine comp_sup_eq_sup_comp g' _ rfl,
  intros f₁ f₂,
  cases f₁,
  { rw [with_bot.none_eq_bot, bot_sup_eq], exact bot_sup_eq.symm, },
  { cases f₂, refl,
    apply congr_arg option.some _,
    exact (hg f₁ f₂), },
end

lemma sup'_apply {s : finset α} (H : s.nonempty) (f : α → Π (b : β), C b) (b : β) :
  s.sup' H f b = s.sup' H (λ a, f a b) :=
comp_sup'_eq_sup'_comp H f (λ x, x b) (λ i j, rfl)
```

Reply:
I guess? I don't see it as a big improvement, but if `comp_sup'_eq_sup'_comp` plugs a hole that someone might miss, I'd be happy to add this (you could either push directly, or I can get to it later).

Reply:
I added the lemma in #7275. It's an improvement because as you have it here, you're proving almost exactly the same statement as `finset.sup'_apply`  without reusing any of the work; `comp_sup'_eq_sup'_comp` proves both `finset.sup_apply'` and `continuous_function.sup'_apply` in one line.

Your `continuous_function.sup'_apply` proof might be a tidier proof than the one used in `finset.sup'_apply` - if so, feel free to replace it in #7275

Reply:
Shall we merge #7275 into this then?

Reply:
Either that or send it off to bors (or both?) - it doesn't really matter to me, I made it separately just in case the scope ended up bigger than I expected. Assuming you go for just merging into this PR, I'll have to find another way to get my 8 commits needed to overtake leanprover-community-bot ;)

Reply:
Oh, sorry, I should have said rather: "shall we make this dependent on #7275 then?".